Iter1

Bloodlog opened the pull request:
Что сделано:
Сервер:
* Есть сервер grpc,
* postgresql с таблицами users, sercrets, tokens,
* jwt авторизация,
* умеет сохранять - шифровать произвольную json строку и дешифровать,
* веб страница со списком клиентов + рекомендуемый клиент(автоопределение ОС пользователя),
* TLS и Генерация сертификатов tls,

Клиент:
Команды сохраняют и использую файл с токеном jwt.
* Регистрация
* Логин
* Прочитать секрет (последняя не удаленная версия секрета) + сохранить в произвольный файл сам секрет,
* Сохранить секрет - json строку,
* Список названий ключей пользователя,
* Удалить секрет (soft delete) + восстановление + удаление метадаты + уничтожение безвозвратное,


Я ориентировался на работу vault-cli клиента от Hashicorp.
Сервис может хранить данные в json формате.


После 1-1:
Добавил команды для загрузки/выгрузки файла на сервер.
minio хранит файл в шифрованном виде, при запросе файла он дешифруется.
Добавил немного юнит тестов.

Bloodlog committed:
iter1 Это первая рабочая версия сервера и клиента для сохранения паролей и другой информации в json формате

Bloodlog committed:
iter1 Это первая рабочая версия сервера и клиента для сохранения паролей и другой информации в json формате

Bloodlog committed:
Merge branch 'main' into iter1

Bloodlog committed:
Попытка исправить линтер

Bloodlog committed:
Версионность ключей и удаление полное + мягкое + метадаты + восстановление

Bloodlog committed:
Переименовал файлы

Bloodlog committed:
Добавил моки для репозиториев

s-shpak commented on internal/proto/v1/model/secret.proto:
```diff
@@ -0,0 +1,30 @@
+edition = "2023";
```
Чтобы начать использовать OpaqueAPI необходимо добавить флаг `api_level`: https://protobuf.dev/reference/go/opaque-migration/

Bloodlog replied:
Исправил

s-shpak commented on internal/handler/vault_grpc_server.go:
```diff
@@ -0,0 +1,179 @@
+	ctx context.Context,
+	req *pbModel.GetSecretRequest,
+) (*pbModel.SecretResponse, error) {
+	userID, err := s.jwtService.GetUserID(req.GetToken())
```
Эту логику лучше реализовать в интерсепторе

Bloodlog replied:
Исправил

Bloodlog committed:
Добавил OpaqueAPI

Bloodlog committed:
Добавил Interceptor

Bloodlog committed:
Добавил тестов на 6%

Bloodlog committed:
Небольшое исправление

Bloodlog committed:
Небольшое исправление тестов

Bloodlog committed:
Добавил тест

Bloodlog committed:
ребилд клиентов

Bloodlog committed:
Добавлены структуры для сохранения файлов

Bloodlog committed:
Добавлены команды для загрузки файлов и выгрузки. Файлы шифруются в minio.

s-shpak reviewed (changes requested):
В целом все очень неплохо. Добавь, пожалуйста, инструкции по запуску в docker-compose, потому что пока я получаю такую ошибку:

```
app-keeper       | exec /app/server: no such file or directory
app-keeper exited with code 1
agent-keeper     | exec /app/agent: no such file or directory
agent-keeper exited with code 1
```

Когда вызываю `docker compose up`

Bloodlog committed:
Попытка исправить билд под линуксом.
Агент теперь не умирает и ждёт команд в docker compose.
Добавил документацию в README по настройке окружения + по загрузке файлов.
Ещё раз протестировал.

Bloodlog commented:
> В целом все очень неплохо. Добавь, пожалуйста, инструкции по запуску в docker-compose, потому что пока я получаю такую ошибку:
> 
> ```
> app-keeper       | exec /app/server: no such file or directory
> app-keeper exited with code 1
> agent-keeper     | exec /app/agent: no such file or directory
> agent-keeper exited with code 1
> ```
> 
> Когда вызываю `docker compose up`

Исправил:
+ Агент теперь не умирает и ждёт команд в docker compose.
+ Добавил документацию в README по настройке окружения + по загрузке файлов.
+ Ещё раз протестировал.


Попробывал исправить билд под линуксом - локально у меня всё запускатся на macos:
+ Попытка исправить билд под линуксом.
